interface-definitions/service_ids_suricata.xml.in

```diff
@@ -0,0 +1,217 @@
+<?xml version="1.0"?>
+<interfaceDefinition>
+  <node name="service">
+    <children>
+      <node name="ids">
+        <properties>
+          <help>Intrusion Detection System</help>
+        </properties>
+        <children>
+          <node name="suricata" owner="${vyos_conf_scripts_dir}/service_ids_suricata.py">
+            <properties>
+              <help>Suricata high performance Network IDS, IPS and Network Security Monitoring engine</help>
+              <priority>740</priority>
+            </properties>
+            <children>
+              #include <include/generic-interface-multi.xml.i>
+              <tagNode name="address-group">
+                <properties>
+                  <help>Address group name</help>
+                  <completionHelp>
+                    <list>home-net external-net http-servers smtp-servers sql-servers dns-servers telnet-servers aim-servers dc-servers dnp3-server dnp3-client modbus-client modbus-server enip-client enip-server</list>
+                  </completionHelp>
+                  <constraint>
+                    <regex>[a-z0-9-]+</regex>
```

Comment:
The `address-group` node lists available completion helpers, but the regex allows much more.

Is it really the case that it's not limited to the ones listed under completionHelp?

Please also add a `valueHelp` XML definition for the ones listed by the completion helper.

Reply:
> Is it really the case that it's not limited to the ones listed under completionHelp?

Yes, the "[Rule-vars]" allow for additional variables to be leveraged in rules; Those for which defaults are defined in this PR are those [provided by default] for Suricata but users can define additional variables to both customize existing groups (e.g., `HOME_NET: "[$LAN_NET,$DMZ_NET]"`) or to support additional custom rules.

The reason I don't want to restrict the `address-group` and `port-group` nodes to those defined by default is that it opens the avenue to support additional rules through [Suricata-Update]. While the current PR only leverages Suricata-Update to fetch the default ET Open rules, future improvements can introduce commands to CRUD other repositories (including private ones) which could require additional variables.

[Rule-vars]: https://docs.suricata.io/en/suricata-6.0.14/configuration/suricata-yaml.html?highlight=address%20group#rule-vars
[Suricata-Update]: https://docs.suricata.io/en/suricata-6.0.14/rule-management/suricata-update.html
[provided by default]: https://github.com/OISF/suricata/blob/7d419c01efdfb2945402675ed220872aca60913b/suricata.yaml.in#L17-L52

Reply:
Okay, I understand what you're trying to accomplish here, having predefined "well known" and custom groups. Why not reflect this in the CLI as is? Something like:

```xml
<node name="address-group">
  <children>
    <leafNode name="predefined">
      <properties>  
        <completionHelp>
          <list>home-net external-net http-servers smtp-servers sql-servers dns-servers telnet-servers aim-servers dc-servers dnp3-server dnp3-client modbus-client modbus-server enip-client enip-server</list>
        </completionHelp>
        <constraint>
          <regex>(home-net|external-net|http-servers|smtp-servers|sql-servers|dns-servers|telnet-servers|aim-servers|dc-servers|dnp3-server|dnp3-client|modbus-client|modbus-server|enip-client|enip-server</list>
        </constraint>
        <multi/>
      </properties>
    </leafNode>
    <tagNode name="custom">
      <properties>
        <help>Address group name</help>
        <valueHelp>
          <format>txt</format>
          <description>Wild Foo Bar one Can do</description>
        <valueHelp>
        <constraint>
          #include <include/constraint/alpha-numeric-hyphen-underscore.xml.i>
        </constraint>
      </properties> 
    </tagNode>
  </children>
</node>
```

My main concern here is that the predefined groupd e.g. `HOME_NET: "[$LAN_NET,$DMZ_NET]"`
are hard to understand from a completion helper standpoint, where is LAN_NET and DMZ_NET defined?

Who defines the addresses for listed in external-net or telnet-servers ?

Reply:
@c-po  `predefined` is not leafNode
Something like this:
```
set service ids suricata address-group <predefined> home-net address 100.64.0.0/24 
set service ids suricata address-group <predefined> home-net address 100.64.1.0/28
...
set service ids suricata address-group <custom> mygroup address 192.0.2.0/24
```

Reply:
First of all, as maintainers, I'm more than happy to accommodate whichever approach you prefer. I don't however believe it would be beneficial to split the values over custom/predefined nodes.

> My main concern here is that the predefined groupd e.g. HOME_NET: "[$LAN_NET,$DMZ_NET]"
are hard to understand from a completion helper standpoint, where is LAN_NET and DMZ_NET defined?

The nodes `LAN_NET` and `DMZ_NET ` are just examples which the user can choose to define as follow:
```
set service ids suricata address-group lan-net address 192.168.1.0/24
set service ids suricata address-group dmz-net address 192.168.2.0/24
set service ids suricata address-group whatever-the-user-wants address 192.168.3.0/24
```
When a user defines a group (address or port), the group can be used both in other groups (e.g., `HOME_NET: "[$LAN_NET,$DMZ_NET,$WHATEVER_THE_USER_WANTS]"`) as well as, most importantly, in Suricata rules (e.g., `alert http $DMZ_NET any -> $LAN_NET any  (msg:"HTTP request from DMZ to internal LAN" ....`).

While the later will be for future improvements (additional conf commands for Suricata update), the former allows advanced users to ease their group management. The above groups can for example generate `HOME_NET: "[$LAN_NET,$DMZ_NET,$WHATEVER_THE_USER_WANTS]"` as follow:
```
set service ids suricata address-group home-net group lan-net
set service ids suricata address-group home-net group dmz-net
set service ids suricata address-group home-net group whatever-the-user-wants
```
It however doesn't mean the user has to define groups, the following example is equivalent for `home-net` definitions.
```
set service ids suricata address-group home-net address 192.168.1.0/24
set service ids suricata address-group home-net address 192.168.2.0/24
set service ids suricata address-group home-net address 192.168.3.0/24
```

> Who defines the addresses for listed in external-net or telnet-servers ?

Suricata provides sane defaults (vars and values) to ensure rules work out of the box. These are the defaults I provided in the definitions (names) and configuration script (values). While users are encouraged to override these with their network's values, the defaults provides already a solid functional base. For example, `telnet-servers` defaults to `$HOME_NET` but using _any_ of the below commands would override the default.
```
set service ids suricata address-group telnet-servers address 192.168.4.0/24
set service ids suricata address-group telnet-servers group some-other-defined-group
```

I'm not really a fan of splitting a similar functionality over `predefined`/`custom` nodes while, in essence, these are similar. The following two entries would either example collide or we would need to add additional restrictions without these having an actual practical need.
```
set service ids suricata address-group predefined home-net address 192.168.1.0/24
set service ids suricata address-group custom home-net address 192.168.1.0/24
```